sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/NotificationHook/WebNotificationHook.cs

```diff
@@ -17,72 +16,65 @@ public partial class WebNotificationHook : NotificationHook
+            CertificateKey = hookParameter.CertificateKey;
+            CertificatePassword = hookParameter.CertificatePassword;
+            Headers = hookParameter.Headers;
         }
 
         /// <summary>
         /// The API address to be called when an alert is triggered.
         /// </summary>
-        public string Endpoint { get => HookParameter.Endpoint; }
+        public string Endpoint { get; set; }
 
         /// <summary>
         /// The username for authenticating to the API address. Leave this blank if authentication isn't needed.
         /// </summary>
-        public string Username { get => HookParameter.Username; set => HookParameter.Username = value; }
+        public string Username { get; set; }
 
         /// <summary>
         /// The password for authenticating to the API address. Leave this blank if authentication isn't needed.
         /// </summary>
-        public string Password { get => HookParameter.Password; set => HookParameter.Password = value; }
+        public string Password { get; set; }
 
         /// <summary>
         /// The certificate key for authenticating to the API address. Leave this blank if authentication isn't needed.
         /// </summary>
-        public string CertificateKey { get => HookParameter.CertificateKey; set => HookParameter.CertificateKey = value; }
+        public string CertificateKey { get; set; }
 
         /// <summary>
         /// The certificate password for authenticating to the API address. Leave this blank if authentication isn't needed.
         /// </summary>
-        public string CertificatePassword { get => HookParameter.CertificatePassword; set => HookParameter.CertificatePassword = value; }
+        public string CertificatePassword { get; set; }
 
         /// <summary>
         /// Custom headers to send in the API call.
         /// </summary>
-        /// <exception cref="ArgumentNullException">The value assigned to <see cref="Headers"/> is null.</exception>
-#pragma warning disable CA2227 // Collection properties should be readonly
-        public IDictionary<string, string> Headers
-        {
-            get => HookParameter.Headers;
-            set
-            {
-                Argument.AssertNotNull(value, nameof(Headers));
-                HookParameter.Headers = value;
-            }
-        }
-#pragma warning restore CA2227 // Collection properties should be readonly
+        public IDictionary<string, string> Headers { get; }
 
         /// <summary>
         /// Used by CodeGen during serialization.
         /// </summary>
-        internal WebhookHookParameter HookParameter { get; private set; }
+        internal WebhookHookParameter HookParameter => new WebhookHookParameter(Endpoint)
+        {
+            Username = Username,
+            Password = Password,
+            CertificateKey = CertificateKey,
+            CertificatePassword = CertificatePassword,
+            Headers = Headers
```

Comment:
should this be `Headers = new Dictionary<string, string>(Headers)` ?

Reply:
I believe there's no reason for instantiating a new dictionary. This property is going to be called during serialization when creating or updating a hook. We get the instance, serialize it, and that's it. We lose the copied reference after that.